cypress/page/products-list.page.ts

```diff
@@ -1,18 +1,25 @@
 class ProductListPage {
     private addToCart: string;
     private ProceedToCheckout: string
+    private productList: string
 
     constructor() {
-        this.addToCart = "#center_column a.button.ajax_add_to_cart_button.btn.btn-default";
+        this.addToCart = "a.button.ajax_add_to_cart_button.btn.btn-default";
         this.ProceedToCheckout = "[style*=\"display: block;\"] .button-container > a"
+        this.productList = "#center_column ul.product_list li"
     }
-    public goAddToCart(): void {
+    public goAddToCart(name: string): void {
+        var product = this.findProductByName(name)
         cy.get(this.addToCart).click()
```

Comment:
You can concatenate the call of the findProductByName with the get, by using `find` function:
`this.findProductByName(productName).find(this.addToCart).click()`

Reply:
Done